src/components/schedule-visualizer/index.tsx

```diff
@@ -217,6 +229,50 @@ export default function ScheduleVisualizer(props: ScheduleVisualizerProps): Reac
   function handleBaseLayerChange(e: L.LayersControlEvent): void {
     debug('set current level name');
     setCurLevelName(e.name);
+    const mapWidth = document.getElementsByClassName('leaflet-image-layer')[0].clientWidth;
+    const mapHeight = document.getElementsByClassName('leaflet-image-layer')[0].clientHeight;
+    setMapDimensions({ width: mapWidth, height: mapHeight });
+    setBound(mapFloorLayers[e.name].bounds);
```

Comment:
A major downside of this is that the zoom in reset each time the floor is changed.